client/src/components/common/HookFormSelect.tsx

```diff
@@ -1,27 +1,27 @@
-import React from "react";
+import React, { FC } from "react";
 
 import PropTypes from "prop-types";
 import { Controller } from "react-hook-form";
 import Select from "react-select";
 
-export const HookFormSelect = ({
-  name,
-  id,
-  control,
-  placeholder,
-  options,
-  required,
-  disabled = false,
-  onChange,
-} = {}) => (
+export const HookFormSelect: FC<{
+  name?: string;
```

Comment:
Here, I would remove the optional assignment (`?:`) from any prop that seems like it's required (I know it's awkward here because you haven't actually used this component... 😬) and also replace the `any`s on the onChange and the control (if possible) with more specific types.

`any` is kind of your TS get-out-of-jail-free card -- useful when you have absolutely no idea what something is, or when it can literally be anything, but is something to avoid otherwise, since you're basically turning off type-checking for anything that is `any`